velox/functions/sparksql/DateTimeFunctions.h

```diff
@@ -951,6 +951,54 @@ struct MillisToTimestampFunction {
   }
 };
 
+template <typename TExec>
+struct SecondsToTimestampFunction {
+  VELOX_DEFINE_FUNCTION_TYPES(TExec);
+
+  template <typename T>
+  FOLLY_ALWAYS_INLINE void call(out_type<Timestamp>& result, T seconds) {
+    if constexpr (std::is_integral_v<T>) {
+      result = Timestamp(static_cast<int64_t>(seconds), 0);
+    } else {
+      // Scale to microseconds using double, truncating toward zero.
+      const double microsD = static_cast<double>(seconds) * 1'000'000.0;
+
+      // Saturate like Java's cast to long.
+      constexpr double kMaxMicrosD =
+          static_cast<double>(std::numeric_limits<int64_t>::max());
+      constexpr double kMinMicrosD =
+          static_cast<double>(std::numeric_limits<int64_t>::min());
+
+      // Cutoff values are based on Java's Long.MAX_VALUE and Long.MIN_VALUE.
+      constexpr int64_t maxSeconds = 9223372036854LL;
+      constexpr int64_t maxNanoseconds = 775807000LL;
+      constexpr int64_t minSeconds = -9223372036855LL;
+      constexpr int64_t minNanoseconds = 224192000LL;
```

Comment:
Can we define above as private static variables instead of declaring them in each call?

Reply:
See my edit. Does this look better?